inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,276 @@
+
+
+class Update(Action):
+
+    def apply_action(self, record, schema, position=0, checked=False):
+        """Recursive function to change a record object."""
+        new_schema, checked, key, where_key, where_negative = self.progress_keys(record, schema, position, checked)
+        if where_negative:  # if the where check was negative return
+            return
+        if not record.get(key):
+            return
+        if position + 1 == len(self.keys):
+            if isinstance(record[key], list):
+                if not self.values_to_check:
+                    record[key] = [self.value if True else x for x in record[key]]
+                if self.values_to_check_regex:
+                    for value_to_check in self.values_to_check:
+                        record[key] = [self.value if re.search(
+                            re.escape(value_to_check),
+                            x) else x for x in record[key]]
+                else:
+                    record[key] = [self.value if x in self.values_to_check else x for x in record[key]]
+                self.changed = True
+            else:
+                if not self.values_to_check\
+                        or record[key] in self.values_to_check:
+                    record[key] = self.value
+                if self.values_to_check_regex:
+                    for value_to_check in self.values_to_check:
+                        if re.search(
+                                re.escape(value_to_check),
+                                record[key]):
+                                record[key] = self.value
+                self.changed = True
+                return
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply_action(array_record, new_schema, position + 1, checked)
+            else:
+                self.apply_action(record[key], new_schema, position + 1, checked)
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
+    new_schema = schema
+    if new_schema['type'] == 'array':
+        new_schema = new_schema['items']['properties']
+    elif new_schema['type'] == 'object':
+        new_schema = new_schema['properties']
+    for key in path:
+        new_schema = new_schema[key]
+        if new_schema['type'] == 'object':
+            new_schema = new_schema['properties']
+            temp_record[key] = {}
+            temp_record = temp_record[key]
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+            if len(path) == 1:
+                temp_record[key] = [value]
+                return record
+            else:
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
```

Comment:
I recommend reading http://okigiveup.net/arguments-against-json-driven-development/ to understand why this kind of code is problematic both in the short term and in the long run.